src/views/TokensListView.js

```diff
@@ -1,86 +1,167 @@
+  const [deleteTokenLoading, setDeleteTokenLoading] = useState(false);
+
+  const handleNewToken = e => {
+    setNewTokenLoading(true);
+    createToken({variables: {name: e.target.name.value}})
+      .then(resp => {
+        refetchTokens().then(resp => setNewTokenLoading(false));
+      })
+      .catch(err => {
+        setNewTokenError(err.message);
+        setNewTokenLoading(false);
+      });
+  };
 
-const TokensListView = () => {
   // onClick event for the delete button next to a token item
-  const onDelete = (callback, name) => {
-    if (
-      window.confirm(
-        'This will break any applications using this token. Are you sure?',
-      )
-    ) {
-      callback({variables: {name}});
-    }
+  const handleDelete = name => {
+    setDeletingToken(name);
+    setConfirmOpen(true);
   };
 
   return (
-    <div className="mx-12 pb-16 BodyContent">
-      <h3 className="mt-0 pt-8 text-blue font-normal">
-        Manage Developer Download Tokens
-      </h3>
-      <p>
-        Developer download tokens allow download of any file using the{' '}
-        <code>?token=</code> query parameter or passed in the Authorization
-        header with a <code>Token</code> prefix.
-      </p>
-      <p>
-        Keep these tokens private! They will not be displayed again after being
-        created.
-      </p>
-      <section className="study-file-list">
-        <Query query={GET_DEV_TOKENS}>
-          {({loading, error, data}) => (
-            <Mutation
-              mutation={DELETE_DEV_TOKEN}
-              update={(cache, {data: {deleteDevToken}}) => {
-                // Removes the token from the allDevTokens query in the cache
-                const {allDevTokens} = cache.readQuery({query: GET_DEV_TOKENS});
-                const deleteIndex = allDevTokens.edges.findIndex(
-                  edge => edge.node.name === deleteDevToken.name,
-                );
-                if (deleteIndex < 0) {
-                  return allDevTokens;
-                }
-                allDevTokens.edges.splice(deleteIndex, 1);
-                const data = {
-                  allDevTokens: {
-                    ...allDevTokens,
-                    edges: allDevTokens.edges,
-                  },
-                };
-                cache.writeQuery({
-                  query: GET_DEV_TOKENS,
-                  data,
-                });
-              }}
-            >
-              {deleteToken => {
-                if (loading)
-                  return <LoadingPlaceholder componentName="Tokens" />;
-                if (error) return `Error!: ${error}`;
-
-                return (
-                  <TokenList
-                    tokens={data.allDevTokens.edges}
-                    deleteToken={name => onDelete(deleteToken, name)}
-                  />
-                );
-              }}
-            </Mutation>
+    <>
+      <Container>
+        <Header as="h3">Manage Developer Download Tokens</Header>
```

Comment:
Maybe add top margins here, and align the header with the rest of content

Reply:
This should be cleared up once the new header is in place.